robot/impl/robot_reconfigure_test.go

```diff
@@ -2579,20 +2578,47 @@ func TestStatusServiceUpdate(t *testing.T) {
 }
 
 func TestRemoteRobotsGold(t *testing.T) {
+	// The test tests that the robot is able to start up with an offline robot,
+	// connect to it and depend on its resources when it comes online,
+	// and react appropriately when it goes offline again. If a separate robot
+	// then comes online again at the same address, the robot should be able to
+	// use their resources as well.
+
+	// To do so, the test initially sets up two remote robots, Remote 1 and 2, and then a third remote, Remote 3,
+	// in the following scenario:
+	// 1) Remote 1's server is started.
+	// 2) The local robot is then set up with resources that depend on resources on both Remote 1 and 2. Since
+	//    Remote 2 is not up, it is expected that some resources will not be built or be available.
```

Comment:
For clarity, we should remove "it is expected that some" and be more straightforward in what we expect:
```
Remote 2 (`bar`) is not up. `bar`s resources are not available to the main robot.
```

Reply:
updated, did not use `bar` though since remote 3 also connects with name `bar`